expression/builtin_math.go

```diff
@@ -516,6 +519,42 @@ func builtinCRC32(args []types.Datum, ctx context.Context) (d types.Datum, err e
 	return d, nil
 }
 
+type signFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *signFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	return &builtinSignSig{newBaseBuiltinFunc(args, ctx)}, errors.Trace(c.verifyArgs(args))
+}
+
+type builtinSignSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinSignSig) eval(row []types.Datum) (types.Datum, error) {
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	return builtinSign(args, b.ctx)
+}
+
+// See http://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_sign
+func builtinSign(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	if args[0].IsNull() {
+		return d, nil
+	}
+	cmp, err := args[0].CompareDatum(ctx.GetSessionVars().StmtCtx, types.NewIntDatum(0))
+	switch {
+	case err == nil:
+	case terror.ErrorEqual(err, types.ErrTruncated):
```

Comment:
Truncate error can't be simply ignored, it's a warning in a select context, but an error in a insert context:

```
mysql> insert into t values (sign('1a'));
ERROR 1292 (22007): Truncated incorrect DOUBLE value: '1a'

mysql> select sign('1a');
+------------+
| sign('1a') |
+------------+
|          1 |
+------------+
1 row in set, 1 warning (0.00 sec)
```